docs/architecture/adr-082-data-companion-api.md

```diff
@@ -0,0 +1,425 @@
+# ADR 082: Data Companion API
+
+## Changelog
+
+- 2022-11-26: Clarify user stories and alternatives, allow for selective
+  publishing of data via the companion API, buffer on disk instead of in memory
+  (@thanethomson)
+- 2022-09-10: First draft (@thanethomson)
+
+## Status
+
+Accepted | Rejected | Deprecated | Superseded by
+
+## Context
+
+This ADR proposes to effectively offload some data storage responsibilities and
+functionality to a **single trusted external companion service** such that:
+
+1. Integrators can expose Tendermint data in whatever format/protocol they want
+   (e.g. JSON-RPC or gRPC).
+2. Integrators can index data in whatever way suits them best.
+3. All users eventually benefit from faster changes to Tendermint because the
+   core team has less, and less complex, code to maintain (implementing this ADR
+   would add more code in the short-term, but would pave the way for significant
+   reductions in complexity in future).
+
+The way the system is currently built is such that a Tendermint node is mostly
+self-contained. While this philosophy initially allowed a certain degree of ease
+of node operation (i.e. simple UX), it has also lent itself to feature sprawl,
+with Tendermint being asked to take care of increasingly more than just
+consensus. Under the spotlight in this ADR are:
+
+1. **Event indexing**, which is required in order to facilitate arbitrary block
+   and transaction querying, as well as subscription for arbitrary events. We
+   have already seen, for instance, how the current (somewhat unreliable) event
+   subscription implementation can cause back-pressure into consensus and affect
+   IBC relayer stability (see [\#6729] and [\#7156]).
+2. **Block execution result storage**, which is required to facilitate a number
+   of RPC APIs, but also results in storage of large quantities of data that is
+   not critical to consensus. This data is, however, critical for certain use
+   cases outside of consensus.
+
+Another intersecting issue is that there may be a variety of use cases that
+require different data models. A good example of this is the mere existence of
+the [PostgreSQL indexer] as compared to the default key/value indexer.
+
+Continuing from and expanding on the ideas outlined in [RFC 006][rfc-006], the
+suggested approach in this ADR is to provide a mechanism to publish certain data
+from Tendermint, in real-time and with certain reliability guarantees, to a
+single companion service outside of the node that can use this data in whatever
+way it chooses (filter and republish it, store it, manipulate or enrich it,
+etc.).
+
+Specifically, this mechanism would initially publish:
```

Comment:
Just in regards to the `initially` part: There isn't any other (global) data that I know of that Tendermint could possibly publish - this is basically everything